src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -165,7 +167,7 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
 
                 // Get the projects from the SolutionManager
                 // Note that projects that are not supported by NuGet, will not show up in this list
-                projects = await _solutionManager.GetNuGetProjectsAsync();
+                projects = (await _solutionManager.GetNuGetProjectsAsync()).ToList();
```

Comment:
why?

Reply:
Because `GetNuGetProjectsAsync` returns an `IEnumerable` and the result, `projects`, will be enumerated more than once.
1. I don't know if an exception might be thrown on the second enumeration
2. Even if it works, I don't know how expensive it is to re-enumerate, whereas saving a list is a fixed cost.

Reply:
We have IEnumerable obsession sometimes :) 

I was only asking why because I saw 1 enumeration and 1 any, which is cheap ;) 

Regardless, that change is ok.